units/commit_queue.go

```diff
@@ -66,6 +66,30 @@ func NewCommitQueueJob(env evergreen.Environment, queueID string, id string) amb
 	return job
 }
 
+func (j *commitQueueJob) TryUnstick(cq *commitqueue.CommitQueue) {
+	//unstuck the queue if the patch is done.
+	nextItem := cq.Next()
```

Comment:
Can you pass in nextItem rather than the whole commit queue?

Reply:
Are you referring to passing it in to the TryUnstick function? I use cq for logging  cq.ProjectID and cq.ProcessingUpdatedTime. I also pass it to the dequeue method.

Reply:
Gotcha--I mostly ask this because potentially between calling cq.Next() the first time (before TryUnstick) and this time we may no longer have a nextItem( if the notification processes it first or it gets manually deleted or etc), so I wanted to work around that. I. think then either pass in nextItem along with the cq object, or check again if nextItem is nil.

Reply:
oh good point! thanks!

Reply:
done